### PR TITLE
feat(core): allow showing build status on commits view

### PR DESCRIPTION
**Pull request by rudoi**

### CODE REVIEW

File: core/src/clients/github.rs [hunk 1]
```diff
@@ -263,4 +268,79 @@ impl GraphQLClient {
...
+                            .history
+                            .nodes
+                            .ok_or(anyhow!("Failed to get valid nodes"))?
+                            .into_iter()
+                            .for_each(|node| {
+                                if let Some(node) = node {
+                                    let oid = node.oid.clone();
+                                    if let Some(status) = node.status {
+                                        let short_oid = oid.chars().take(8).collect::<String>();
+                                        match status.state {
```
Comment (rdunnington) [inline core/src/clients/github.rs:314]: could compress this a bit with 
```rust
let status_str = match status.state {
get_commit_statuses::StatusState::SUCCESS => "success",
// etc
};
map.insert(short_oid, status_str.to_string());
```

**Review by rdunnington — APPROVED** [review]
left 1 minor suggestion

nice work

File: core/src/clients/github.rs (hunk 1 above)
Comment (rudoi) [reply core/src/clients/github.rs]: done!

File: core/src/clients/github.rs [hunk 2]
```diff
@@ -263,4 +268,73 @@ impl GraphQLClient {
...
+                                    let oid = node.oid.clone();
+                                    if let Some(status) = node.status {
+                                        let short_oid = oid.chars().take(8).collect::<String>();
+                                        let status_str: &str = match status.state {
+                                            get_commit_statuses::StatusState::SUCCESS => "success",
+                                            get_commit_statuses::StatusState::FAILURE => "failure",
+                                            get_commit_statuses::StatusState::ERROR => "error",
+                                            get_commit_statuses::StatusState::EXPECTED => {
+                                                "expected"
+                                            }
```
Comment (rdunnington) [inline core/src/clients/github.rs:320]: weird that this one is formatted differently than the others :P

**Review by rdunnington — APPROVED** [review]

File: core/src/clients/github.rs (hunk 2 above)
Comment (rudoi) [reply core/src/clients/github.rs:320]: it's longer :)

**Review by sylviacx — APPROVED** [review]

**Pull request merged**